### DISCUSSION TITLE
Cool project

### DISCUSSION THREAD

**Discussion by jjl** [lunatic-solutions/lunatic#15] [General]
Hi, I just got linked to your project, and I think it's pretty cool.

I'm mostly an elixir programmer but i have some async rust projects i hack on when i have time, including [async-backplane](https://github.com/irrustible/async-backplane), which i think captures the essential soul of erlang's error handling mechanism, although it's an incredibly minimal version of it because I don't as much time to spend hacking rust as I'd like.

Lunatic (or something awfully like it) has been a future plan of mine for as and when I get the time, so i'm pumped someone else has gone and done the hard work to get things started. I'd love to get involved as I have the time, though I haven't been keeping up with wasm developments as much as I'd have liked this year.

As I recall, there wasn't much flexibility in the ABI when I last checked - you had to manually monomorphise everything and such. Have things improved on that front?

Best,
James

**Comment by bkolobara** [comment]
Hi @jjl,

thanks, I would love to bring other contributors onto the project. I just want to merge some big architectural changes I have been working on for the last few weeks and write some documentation before I do so.

There are many improvements landing soon. Just drop me an email at me@kolobara.com and I will make sure to get back to you and assist you to get started with hacking on Lunatic.

**Comment by bkolobara** [comment]
The Lunatic repository has Github Discussions enabled: https://github.com/lunatic-lang/lunatic/discussions.

This is issue would be a great example of a discussion. I will close it here and we can continue the topic there.

**Comment by zicklag** [comment]
@bkolobara Just a little administration note, you can actually convert this issue to a discussion with a button that is in the bottom of the right sidebar if you want so that a new discussion doesn't have to be manually opened.

**Reply by bkolobara** [reply]
Thanks a lot for the tip! I didn't realise this was an option.
